src/main/java/com/redhat/devtools/lsp4ij/features/completion/LSPCompletionTriggerTypedHandler.java

```diff
@@ -1,74 +1,76 @@
-            // The current typed character is defined as "completion trigger characters"
-            // in a associated language server, open the completion.
-            AutoPopupController.getInstance(project).scheduleAutoPopup(editor);
-            result = Result.STOP;
-        }
-
-        if (hasLanguageServerSupportingSignatureTriggerCharacters(charTyped, project, file)) {
-            // The current typed character is defined as "signature trigger characters"
-            // in a associated language server, open the signature help.
-            AutoPopupController.getInstance(project).autoPopupParameterInfo(editor, null);
-            result = Result.STOP;
-        }
-
-        return result;
-    }
-
-    private static boolean hasLanguageServerSupportingCompletionTriggerCharacters(char charTyped, Project project, PsiFile file) {
-        return LanguageServiceAccessor.getInstance(project)
-                .hasAny(file.getVirtualFile(), ls -> ls.getClientFeatures()
-                        .getCompletionFeature()
-                        .isCompletionTriggerCharactersSupported(file, String.valueOf(charTyped)));
-    }
-
-    private static boolean hasLanguageServerSupportingSignatureTriggerCharacters(char charTyped, Project project, PsiFile file) {
-        return LanguageServiceAccessor.getInstance(project)
-                .hasAny(file.getVirtualFile(), ls -> ls.getClientFeatures()
-                                .getSignatureHelpFeature()
-                                .isSignatureTriggerCharactersSupported(file, String.valueOf(charTyped)));
-    }
-}
+/*******************************************************************************
+ * Copyright (c) 2024 Red Hat, Inc.
+ * Distributed under license by Red Hat, Inc. All rights reserved.
+ * This program is made available under the terms of the
+ * Eclipse Public License v2.0 which accompanies this distribution,
+ * and is available at http://www.eclipse.org/legal/epl-v20.html
+ *
+ * Contributors:
+ * Red Hat, Inc. - initial API and implementation
+ ******************************************************************************/
+package com.redhat.devtools.lsp4ij.features.completion;
+
+import com.intellij.codeInsight.AutoPopupController;
+import com.intellij.codeInsight.editorActions.TypedHandlerDelegate;
+import com.intellij.openapi.editor.Editor;
+import com.intellij.openapi.project.Project;
+import com.intellij.psi.PsiFile;
+import com.redhat.devtools.lsp4ij.LanguageServersRegistry;
+import com.redhat.devtools.lsp4ij.LanguageServiceAccessor;
+import org.jetbrains.annotations.ApiStatus;
+import org.jetbrains.annotations.NotNull;
+
+/**
+ * {@link TypedHandlerDelegate} implementation used to support LSP
+ * completion triggerCharacters to open an IJ completion as popup
+ * when language server defines the current typed character as completion trigger characters in the
+ * server capability.
+ */
+public class LSPCompletionTriggerTypedHandler extends TypedHandlerDelegate {
```

Comment:
Renamed this to be more specific about its role now that there are multiple `typedHandler` EP implementations in the plugin.